sdk/azidentity/interactive_browser_credential.go

```diff
@@ -0,0 +1,150 @@
+// options: allow to configure the management of the requests sent to Azure Active Directory.
+func NewInteractiveBrowserCredential(options *InteractiveBrowserCredentialOptions) (*InteractiveBrowserCredential, error) {
+	var credentialOptions *TokenCredentialOptions
+	tenantID := "organizations"
+	clientID := developerSignOnClientID
+	redirectURI := ""
+	var clientSecret string
+	if options != nil {
+		if options.Options != nil {
+			credentialOptions = options.Options
+		}
+		if options.TenantID != nil {
+			tenantID = *options.TenantID
+		}
+		if options.ClientID != nil {
+			clientID = *options.ClientID
+		}
+		if options.ClientSecret != nil {
+			clientSecret = *options.ClientSecret
+		}
+		if options.RedirectURI != nil {
+			redirectURI = *options.RedirectURI
+		}
+	}
+	c, err := newAADIdentityClient(credentialOptions)
+	if err != nil {
+		return nil, err
+	}
+	return &InteractiveBrowserCredential{tenantID: tenantID, clientID: clientID, clientSecret: clientSecret, redirectURI: redirectURI, client: c}, nil
+}
+
+// GetToken obtains a token from Azure Active Directory using an interactive browser to authenticate.
+// ctx: Context used to control the request lifetime.
+// opts: TokenRequestOptions contains the list of scopes for which the token will have access.
+// Returns an AccessToken which can be used to authenticate service client calls.
+func (c *InteractiveBrowserCredential) GetToken(ctx context.Context, opts azcore.TokenRequestOptions) (*azcore.AccessToken, error) {
+	tk, err := c.client.authenticateInteractiveBrowser(ctx, c.tenantID, c.clientID, c.clientSecret, c.redirectURI, opts.Scopes)
+	if err != nil {
+		addGetTokenFailureLogs("Interactive Browser Credential", err, true)
+		return nil, err
+	}
+	logGetTokenSuccess(c, opts)
+	return tk, nil
+}
+
+// AuthenticationPolicy implements the azcore.Credential interface on InteractiveBrowserCredential and calls the Bearer Token policy
+// to get the bearer token.
+func (c *InteractiveBrowserCredential) AuthenticationPolicy(options azcore.AuthenticationPolicyOptions) azcore.Policy {
+	return newBearerTokenPolicy(c, options)
+}
+
+var _ azcore.TokenCredential = (*InteractiveBrowserCredential)(nil)
+
+var authCodeReceiver = func(authorityHost string, tenantID string, clientID string, redirectURI string, scopes []string) (*interactiveConfig, error) {
+	return interactiveBrowserLogin(authorityHost, tenantID, clientID, redirectURI, scopes)
+}
+
+// interactiveBrowserLogin opens an interactive browser with the specified tenant and client IDs provided then returns the authorization code
+// received or an error.
+func interactiveBrowserLogin(authorityHost string, tenantID string, clientID string, redirectURL string, scopes []string) (*interactiveConfig, error) {
+	const authURLFormat = "%s/%s/oauth2/v2.0/authorize?response_type=code&response_mode=query&client_id=%s&redirect_uri=%s&state=%s&scope=%s&prompt=select_account"
+	if tenantID == "" {
+		tenantID = "organizations"
```

Comment:
Didn't this already get set elsewhere?

Reply:
Same comment applies to `clientID` below.

Reply:
Done